src/components/elements/table/TableRowActions.tsx

```diff
@@ -21,35 +25,62 @@ const TableRowActions = <T extends { id: string }>({
   primaryActionConfig,
   secondaryActionConfigs,
 }: TableRowActionsProps<T>) => {
+  const isTiny = useIsMobile('sm');
+
+  const primaryAria =
+    primaryActionConfig?.ariaLabel ||
+    `${primaryActionConfig?.title}, ${recordName}`;
+
+  const buttonProps: Pick<ButtonProps, 'size' | 'variant' | 'aria-label'> = {
+    size: 'small',
+    variant: 'outlined',
+    'aria-label': primaryAria,
+  };
+
   return (
-    <Stack direction='row' alignItems='center' justifyContent='end' gap={0.5}>
-      {!!primaryActionConfig && primaryActionConfig.to && (
-        <ButtonLink
-          to={primaryActionConfig.to || ''}
-          size='small'
-          variant='outlined'
-          aria-label={
-            primaryActionConfig.ariaLabel ||
-            `${primaryActionConfig.title}, ${recordName}`
-          }
-          state={primaryActionConfig.linkState}
-        >
-          {primaryActionConfig.title}
-        </ButtonLink>
-      )}
-      {!!primaryActionConfig && primaryActionConfig.onClick && (
-        <Button
-          onClick={primaryActionConfig.onClick}
-          size='small'
-          variant='outlined'
-          aria-label={
-            primaryActionConfig.ariaLabel ||
-            `${primaryActionConfig.title}, ${recordName}`
-          }
-        >
-          {primaryActionConfig.title}
-        </Button>
-      )}
+    <Stack
+      direction='row'
+      alignItems='center'
+      justifyContent='end'
+      gap={{ xs: 0, sm: 0.5 }}
+    >
+      {!!primaryActionConfig &&
+        primaryActionConfig.to &&
+        (isTiny ? (
+          <ButtonTooltipContainer title={primaryAria}>
+            <ButtonLink
+              to={primaryActionConfig.to || ''}
+              state={primaryActionConfig.linkState}
+              {...buttonProps}
+              variant='text'
+              sx={{ maxWidth: '20px', minWidth: 0 }}
+            >
+              <GoIcon />
+            </ButtonLink>
+          </ButtonTooltipContainer>
+        ) : (
+          <ButtonLink
+            to={primaryActionConfig.to || ''}
+            state={primaryActionConfig.linkState}
+            {...buttonProps}
+          >
+            {primaryActionConfig.title}
+          </ButtonLink>
+        ))}
+      {!!primaryActionConfig &&
+        primaryActionConfig.onClick &&
+        (isTiny ? (
+          <IconButtonContainer
+            Icon={GoIcon}
+            onClick={primaryActionConfig.onClick}
+            tooltip={primaryAria}
+            IconButtonProps={{ size: 'medium' }}
+          />
+        ) : (
+          <Button onClick={primaryActionConfig.onClick} {...buttonProps}>
+            {primaryActionConfig.title}
+          </Button>
+        ))}
       {primaryAction}
       {!!secondaryActionConfigs && secondaryActionConfigs.length > 0 && (
```

Comment:
Noting that I think there was agreement around including the primary action config inside the secondary menu, so I'd say we can go ahead with that. Maybe it would be as simple as passing `items={[primaryActionConfig , ...secondaryActionConfigs]}` below.